src/main/java/util/TickingTimerManager.java

```diff
@@ -0,0 +1,87 @@
+package util;
+
+/**
+ * This class manages the TickingTimer instance used for refreshing. It computes the refresh rate period and change
+ * the refresh period.
+ */
+public class TickingTimerManager {
+
+
+    public static final int BUFFER_TIME = 1; //To allow some additional time until next refresh.
+    private final TickingTimer timer;
+    public static final int REFRESH_PERIOD = 60;
+    //The amount that is set aside for manual refresh or creation of issues, labels and milestones.
+    public static final int APIQUOTA_BUFFER = 200;
+
+    public TickingTimerManager(TickingTimer timer) {
+        this.timer = timer;
+    }
+
+    public void startTimer() {
+        timer.start();
+    }
+
+    public void restartTimer() {
+        timer.restart();
+    }
+
+    /**
+     * Computes the TickerTimer period that are used for refreshing the issues periodically.
+     * @param apiQuota         : The remaining allowed api request until the next api request allowance renewal.
+     * @param remainingTime    : The remaining time left until the next api request allowance renewal.
+     * @param lastApiCallsUsed : The amount of api used in the last api pull.
+     */
+    public long computeTickerTimerPeriod(int apiQuota, long remainingTime, int lastApiCallsUsed) {
+
+        assert apiQuota >= 0 && remainingTime >= 0 && lastApiCallsUsed >= 0;
+
+        long refreshTimeInMins = 1;
+
+        if (isDuringAppInitOrRemainingTimeIsZero(apiQuota, remainingTime, lastApiCallsUsed)) {
+            refreshTimeInMins = Utility.secsToMins(REFRESH_PERIOD);
+            return refreshTimeInMins;
+        }
+
+        if (isQuotaInsufficient(apiQuota, lastApiCallsUsed)) {
+            refreshTimeInMins = remainingTime + BUFFER_TIME;
+            return refreshTimeInMins;
+        }
+
+        if (isQuotaSufficient(apiQuota, lastApiCallsUsed)) {
+
+            refreshTimeInMins = (long) Math.ceil(remainingTime / computeNoOfRefreshAllowed(apiQuota, lastApiCallsUsed));
+
+            if (computeNoOfRefreshAllowed(apiQuota, lastApiCallsUsed) == 1) {
+                refreshTimeInMins++;
+            }
+            return refreshTimeInMins;
+        }
+
+        return refreshTimeInMins;
```

Comment:
So the method returns 1 if none of the conditions are true?
In that case just `return 1;`